test/variable.jl

```diff
@@ -14,6 +14,21 @@ using JuMP
 import JuMP.repl
 using Base.Test
 
+mutable struct MyVariable
+    haslb::Bool
+    lowerbound::Number
+    hasub::Bool
+    upperbound::Number
+    hasfix::Bool
```

Comment:
The documentation in `macros.jl` of `constructvariable!` is not in sync with this (it doesn't mention `haslb`, `hasub`, `hasfix`, `fixedvalue`, or `hasstart`). This whole interface with positional arguments is very confusing. One possible improvement is to have the macro pass a `VariableInfo` struct to `constructvariable!` where `VariableInfo` is defined as
```julia
struct VariableInfo
    haslb::Bool
￼    lowerbound::Number
￼    hasub::Bool
￼    upperbound::Number
￼    hasfix::Bool
    fixedvalue::Number
￼    binary::Bool
￼    integer::Bool
￼    name::String
￼    hasstart::Bool
￼    start::Number
end
```
(maybe parameterized on the number type for performance?)
This way the keyword arguments are separated from "special" macro syntax.

Reply:
Done :)